lib/block/blockdownload.js

```diff
@@ -112,40 +89,127 @@ class BlockDownload extends EventEmitter {
+            this.emit("filestart", transferEvent.transferAsset.eventData);
+        });
+
+        controller.on(TransferEvents.JOIN_TRANSFER_PARTS, transferEvent => {
+            this.emit("fileprogress", {
+                ...transferEvent.transferAsset.eventData,
+                transferred: transferEvent.props.transferBytes
+            });
+        });
+
+        controller.on(TransferEvents.AFTER_JOIN_TRANSFER_PARTS, transferEvent => {
+            this.emit("fileend", transferEvent.transferAsset.eventData);
+        });
+
+        controller.on(TransferEvents.ERROR, transferEvent => {
+            console.log(`Error during block download: ${transferEvent.error}`);
+            this.errorEvents.push(transferEvent);
+
+            if (transferEvent.props.firstError) {
+                this.emit("fileerror", {
+                    ...transferEvent.transferAsset.eventData,
+                    errors: [DownloadError.fromError(transferEvent.error)]
+                });
+            }
+        });
+
+        console.log("Block download: block download controller created");
+        return controller;
+    }
+
+    /**
+     * Removes all listeners for events 
+     * `CREATE_TRANSFER_PARTS`, `JOIN_TRANSFER_PARTS`,
+     * `AFTER_JOIN_TRANSFER_PARTS` and `ERROR`
+     * from a block download controller
+     * @param {TransferController} controller a block download controller
+     */
+    finalizeController(controller){
+        if(!controller){
+            return;
+        }
+        
+        try {
+            controller.removeAllListeners(TransferEvents.CREATE_TRANSFER_PARTS);
+            controller.removeAllListeners(TransferEvents.JOIN_TRANSFER_PARTS);
+            controller.removeAllListeners(TransferEvents.AFTER_JOIN_TRANSFER_PARTS);
+            controller.removeAllListeners(TransferEvents.ERROR);
+        } catch (err) {
+            console.log(`Failed to remove event listeners from block download controller: ${err}`);
+        }
+    }
+
     /**
      * Download files to local disk
      * Throws the first unrecoverable error if unsuccessful, all others are logged
      * 
      * @param {BlockDownloadOptions} options Block download options
      */
-    async downloadFiles (options = {}) {
+    async downloadFiles(options = {}) {
         const preferredPartSize = options.preferredPartSize || DEFAULT_PART_SIZE;
         const maxConcurrent = options.maxConcurrent || DEFAULT_MAX_CONCURRENCY;
-        const controller = createController();
 
         const retryOptions = {
-            retryMaxCount: options.retryCount || DEFAULT_RETRY_COUNT
+            retryMaxCount: options.retryCount || DEFAULT_RETRY_COUNT,
+            timeout: options.timeout,
+            retryAllErrors: options.retryAllErrors,
+            retryMaxDuration: options.retryMaxDuration,
+            retryInitialDelay: options.retryInitialDelay,
+            retryBackoff: options.retryBackoff
```

Comment:
I went with using the filter function in `file.js` to pass along the options: https://github.com/adobe/node-httptransfer/blob/6e588e161cb47541d5b5d13aa5a2db6522631e02/lib/file.js#L156-L161
this is what we did for the older functions in the past: https://github.com/adobe/node-httptransfer/blob/6e588e161cb47541d5b5d13aa5a2db6522631e02/lib/file.js#L285-L289

another reason I did it that way for upload is because the options are not just retry options. for example, `method` was passed to `options` to use `POST` instead of `PUT` for upload.

Reply:
Download will always be `GET`. The method here can't change.

Reply:
also re my comment here: https://github.com/adobe/node-httptransfer/pull/81/files#r773328617
upload is missing `retryBackoff` and `retryInitialDelay`
and download is missing `retryEnabled`

I can do this in a follow up PR

Reply:
@tmathern yeah that makes sense so in the case of download, it will always only be retry options. I think we should filter the options in the same place in both upload/download to keep consistency. wdyt?

Reply:
Sure, but let me rather do all this in the refactoring branch, so it's all nice and consistent.